airbyte-commons/src/main/java/io/airbyte/commons/json/Jsons.java

```diff
@@ -189,6 +198,25 @@ public static int getIntOrZero(final JsonNode json, final List<String> keys) {
     return optional.map(JsonNode::asInt).orElse(0);
   }
 
+  public static JsonNode flattenConfig(final JsonNode config) {
+    return flattenConfig((ObjectNode) Jsons.emptyObject(), (ObjectNode) config);
+  }
+
+  private static ObjectNode flattenConfig(final ObjectNode flatConfig, final ObjectNode configToFlatten) {
+    final List<String> keysToFlatten = new ArrayList<>();
+    for (final String key : Jsons.keys(configToFlatten)) {
+      if (configToFlatten.get(key).getNodeType() == OBJECT) {
+        keysToFlatten.add(key);
+      } else if (!flatConfig.has(key)) {
+        flatConfig.set(key, configToFlatten.get(key));
+      } else {
+        LOGGER.warn(String.format("Config's key '%s' already exists", key));
```

Comment:
just noticed this - I think this is too risky. This function could fail without the developer ever realizing. I think we should put this method only in the oauth context where we know it is safe, and even then , error on this else statement as it breaks an assumption the developer is making

it's also not a fully generic flatten function, which is another reason I'd advocate for putting it only in the oauth context instead of Jsons.java

Reply:
👌 I'm moving it to an OAuth specific class and throw exceptions in case the config is not properly formed